src/components/Article/Article.tsx

```diff
@@ -13,36 +13,49 @@ type ArticleProps = {
 };
 
 const Article = ({ resId, classes }: ArticleProps): typeof Article => {
-  const resources = useSelector((state: RootState) => state.resources);
-  const article = resources[resId];
-  const sections = useSelector((state: RootState) => state.sections);
-  const calcProgress = () => {
-    return 25;
-  };
-
-  if (isArticle(article.data)) {
-    const articleSections = article.data.sections;
-    return (
-      <div>
-        <div className={classes.title}>
-          <Typography variant="h1">{article.title}</Typography>
-        </div>
-        <div className={classes.progressBar}>
-          {ProgressBar(calcProgress())}
-          <Typography variant="body2" className={classes.progressText}>
-            % Complete
-          </Typography>
-        </div>
-        <div className={classes.sectionView}>
-          {articleSections.map((key: any) => (
-            <div className={classes.section}>
-              <SectionCard section={sections[key]} />
-            </div>
-          ))}
-        </div>
-      </div>
-    );
-  }
-};
+    const resources = useSelector((state: RootState) => state.resources)
+    const article = resources[resId]
+    const sections = useSelector((state: RootState) => state.sections)
+
+    const [scrollTop, setScrollTop] = useState(0)
+    
+    const onScroll = () => {
+      const winScroll = document.documentElement.scrollTop
+      const height = document.documentElement.scrollHeight - document.documentElement.clientHeight
+
+      const scrolled = (winScroll / height) * 100
+
+      setScrollTop(scrolled)
+    }
+
+    useEffect(() => {
+      window.addEventListener("scroll", onScroll)
+
 
-export default withStyles(styles)(Article);
+      return () => window.removeEventListener("scroll", onScroll)
+    }, [])
+    
+    if (isArticle(article.data)) {
+        const articleSections = article.data.sections
+        return (
+            <div>
+              <div className={classes.fixed}>
+                <div className={classes.title}>
+                  <Typography variant="h1">{article.title}</Typography>
+                </div>
+                <div className={classes.progressBar}>
+                  {ProgressBar(scrollTop, 'article')}
+                </div>
+                <div className={classes.progressText}>
+                  <Typography variant="h3">{Math.round(scrollTop)} % <br /> Complete</Typography>
```

Comment:
I think the break in the percent and the complete might be because of Fang adjusting font sizes! (One line is fine)